charts/metering-ansible-operator/templates/crds/report.crd.yaml

```diff
@@ -35,3 +35,150 @@ spec:
   - name: Age
     type: date
     JSONPath: .metadata.creationTimestamp
+  validation:
+    openAPIV3Schema:
+      type: object
+      required:
+      - spec
+      properties:
+        # note: see https://github.com/operator-framework/operator-metering/blob/master/Documentation/reports.md for more information
+        # note: apiVersion, kind, metadata is implicitly validated by API server
+        # note: do we want to add `default: 0` to the minute/second/hour etc.
+        spec:
+          type: object
+          required:
+          - query
+          properties:
+            # there's no way to enforce what reportqueries we can enforce (besides the default ones using the `enum` array keyword)
+            # due to roll-up reports, custom reportqueries, etc.
+            query:
+              type: string
+              minLength: 1
+
+            reportingStart:
+              type: string
+              format: date
+
+            reportingEnd:
+              type: string
+              format: date
+
+            runImmediately:
+              type: boolean
+
+            inputs:
+              type: array
+              minItems: 1
+              items:
+                type: object
+                required:
+                - name
+                - value
+                properties:
+                  name:
+                    type: string
+                    minLength: 1
+                  value:
+                    type: string
```

Comment:
This should be a `oneOf` type I believe, the value can be any type type: eg array, object, string, double.

Reply:
hmm I wonder if I can just use `AnyValue: {}` instead of specifying each of the types explicitly using `oneOf`.

Reply:
I guess I could always just define the `value` property as `value: {}` as `AnyValue` is implicitly defined.